### ISSUE TITLE
[Issue]: Audio/Subtitle track localisation problems

### ISSUE THREAD

**Issue by jmercz** [jellyfin/jellyfin#12272]
### Please describe your bug

Hello everyone,

I figured that the localisation of the Audio/Subtitle track selection is not fully complete and not quite consistent in the sense of which settings are set at different locations. This issue is a follow up of a forum post I posted into 'Troubleshooting', since I found another problem that is described below and wanted to discuss the general handling of the localisation of the track selection. I link the post [here](https://forum.jellyfin.org/t-solved-localisation-of-track-selection).

My Jellyfin Server (10.9.7) is running as a Docker Container, drawing the latest image, on a Synology NAS with DSM 2.7.1.
A more detailed description of setup and outcome (with pictures) is given in the "Reproduction Steps" part.

### Initial Setup
Starting from an all English setup with the following GUI options:
- Administration > Dashboard > General > Preferred display language: English (United Kingdom)
- User > Settings > Display > Display language: English (United Kingdom)

and the following environment variables set for the Docker container:
```
LC_ALL=en_UK.UTF-8
LANG=en_UK.UTF-8
LANGUAGE=en
```

The Audio/Subtitle track selection looks as follows:
![1_server-en_gui-en_environment-en](https://github.com/user-attachments/assets/892eda48-ef1e-43de-baf5-aec50a85e4df)
Everything in English, **as expected**. As far as I can tell, the tracks (options in the dropdown menu) are constructed from metadata flags of the respective tracks in the .mkv file.

### Changing to Localisation

Changing the user language under User > Settings > Display > Display language to _German_, the track selection looks like:
![2_server-en_gui-de_environment-en](https://github.com/user-attachments/assets/46bfe24c-0382-4de5-8b84-ae0d5491efb2)
Note that the categories on the left are indeed translated correctly (_Subtitles_ changed to _Untertitel_, _Video_ and _Audio_ is German as well. The selections/dropdowns remain English.

Now, changing the server preferred display language (Administration > Dashboard > General > Preferred display language) to _German_ as well:
![3_server-de_gui-de_environment-en](https://github.com/user-attachments/assets/aad3a3cd-a5bc-4a99-beb0-35d95b30f6ac)
Note, that only the default flag of the subtitle track got translated (from _Default_ to _Standard_). But **only** for the subtitle track.

Going even deeper into server settings by stopping the server, changing the environment variables to:
```
LC_ALL=de_DE.UTF-8
LANG=de_DE.UTF-8
LANGUAGE=de:en
```
and restarting the server, the selection looks like:
![4_server-de_gui-de_environment-de](https://github.com/user-attachments/assets/1b02aa72-f0ff-4f48-9a31-df9309467422)
now successfully translating the language flag (_German_ to _Deutsch_) for both tracks. However, the default flag for the audio track **still is not translated**. (_Surround_ and the codecs are proper nouns and don't have to be translated)

### Summary
There seems to be an issue with default flags not beeing translated for audio tracks - unintentionally translating it in the subtitle tracks, but I assume the former - which requires a fix.

Furthermore, I'd like to discuss the general handling of the localisation of the track selection menu. In my humble opinion, the track selection is an integral part of the GUI that is presented to the user. In fact, the user interacts with that very part of the GUI. Therefore, I think the localisation of the track selection should comply with and only with the selected user language (User > Settings > Display > Display language). Having to change options in three locations, that all alter different aspects, is unintuitive and prone to errors. Moreover, a restricted user with different language preferrences might not have access to the server settings, but should be able to customise his experience to his desires.

In conclusion, I hope I did not sound too harsh. I am very much delighted by Jellyfin and have the highest of respect for all the contributors who sacrifice their valuable free time to create something brilliant that we can all enjoy. For Free!
What are your thoughts about this?

Best regards,
Julius

### Jellyfin Version

10.9.0

### Environment

```markdown
- OS: DSM 7.2.1-69057 Update 5
- Linux Kernel: unknown
- Virtualization: Docker
- Clients: Browser, Android, Android TV
... rest not applicable
```

### Jellyfin logs

```shell
No actions are taken, thus no log file should be of any use here.
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by felix920506** [comment]
Duplicate of #6302

**Comment by jmercz** [comment]
@felix920506 
> Duplicate of #6302

I think I might have created some misunderstanding here. As oposed to #6302, the language tags are correctly recognised. My concern is the influence of multiple settings on the localisation of the track selection and the lack of translation for a certain tag. Therefore I beg to differ that it is a duplicate.
Or is that and the moving to the Triage Board common practice for further investigation? If so, I'm very sorry for the hassle. First time for me, contributing to such a project.

**Comment by felix920506** [comment]
Yeah you are right, this is indeed a different issue, reopening this one.

**Comment by felix920506** [comment]
Upon closer examination, it seems like this issue is reporting multiple issues in one. Please open new issues for each one, thanks. 

1. subtitle selector and audio selector "Default" part behaving differently to language settings

2. Should we display track languages in original language or client language (should we display "日本語" or "Japanese" for a Japanese track if language is set to English)

3. The selector should be made into a client string not a server string for better localization support.

**Comment by jmercz** [comment]
@felix920506 
> Upon closer examination, it seems like this issue is reporting multiple issues in one. Please open new issues for each one, thanks.
> 
>     1. subtitle selector and audio selector "Default" part behaving differently to language settings
> 
>     2. Should we display track languages in original language or client language (should we display "日本語" or "Japanese" for a Japanese track if language is set to English)
> 
>     3. The selector should be made into a client string not a server string for better localization support.

Thanks for taking care of the issue.
Recommendation 1 is in issue #12276, recommendation 2 and 3 in #12277. I think that language flags are being localised according to settings is an expected behaviour. Since the user already sets a language when setting up his client, he will never see an "unlocalised" track selector, thus the language flag does not neet to be in its "native speaker form". I hope that was understandable now. :P
